src/mainwindow/mainwindow.cpp

```diff
@@ -63,31 +67,170 @@ MainWindow::~MainWindow()
 // SLOTS
 ///////////////////////////////////////////////////////////
 
+void MainWindow::on_updateWidget_clicked()
+{
+    QByteArray testing_array;
+    testing_array.resize(29); // https://blog.fireheart.in/a?ID=01750-818441ff-32e9-419f-9738-f66b0570d4d2
+
+    testing_array[0] = 0x7E;
+    testing_array[1] = 0x00;
+    testing_array[2] = 0x19;
+    testing_array[3] = 0x90;
+    testing_array[4] = 0x01;
+    testing_array[5] = 0x00;
+    testing_array[6] = 0x13;
+    testing_array[7] = 0xA2;
+    testing_array[8] = 0x00;
+    testing_array[9] = 0x41;
+    testing_array[10] = 0xB1;
+    testing_array[11] = 0x6D;
+    testing_array[12] = 0x1C;
+    testing_array[13] = 0xFF;
+    testing_array[14] = 0xFE;
+    testing_array[15] = 0x00;
+    testing_array[16] = 0x00;
+    testing_array[17] = 0x53;
+    testing_array[18] = 0x45;
+    testing_array[19] = 0x4E;
+    testing_array[20] = 0x54;
+    testing_array[21] = 0x20;
+    testing_array[22] = 0x46;
+    testing_array[23] = 0x52;
+    testing_array[24] = 0x4F;
+    testing_array[25] = 0x4D;
+    testing_array[26] = 0x20;
+    testing_array[27] = 0x42;
+    testing_array[28] = 0xD1;
+
+    updateWidget(testing_array);
+}
+
 /**
  * @brief Slot to update the GUI with data received from the plane; for CV data, it also outputs to the POGI file
  *
  * @param encoded_msg The complete message as parsed by the handleSerialRead signal
  *
  */
-void MainWindow::updateWidget(QByteArray encoded_msg)
+void MainWindow::updateWidget(QByteArray encoded_msg) // ex: 7E | 00 19 | 10 | 01 | 00 13 A2 00 41 B1 6D 1C | FF FE | 00 | 00 | 53 45 4E 54 20 46 52 4F 4D 20 42 | D1
 {
-
     /* decode the data */
 
-    mavlink_decoding_status_t decoderStatus = MAVLINK_DECODING_INCOMPLETE;
-
-    char decoded_message_buffer[50]; //256 is the max payload length
-
     // decoder gets one byte at a time from a serial port
     POGI_Message_IDs_e message_type = POGI_MESSAGE_ID_NONE;
 
-    for( int i = 0; i < encoded_msg.size(); i++) // 50 is just a random number larger than message length (for GPS message length is 39)
+    // constants
+    const uint STARTING_DELIMINATOR_7E_VALUE = 126;
+    const uint FRAME_TYPE_90_VALUE = 144;
+    const uint FRAME_TYPE_8B_VALUE = 139;
+
+    // Double check correct data types in declaration
+    quint8 start_deliminator = encoded_msg[0]; // unsigned byte 7E
+    qDebug() << "Start Deliminator: ";
+    qDebug() << start_deliminator; // outputs 126
+
+    quint8 frame_length = encoded_msg[2]; // READ THE LENGTH OF FRAME
+    qDebug() << "Frame length: ";
+    qDebug() << frame_length; // outputs 25
+    uint length_of_message = frame_length - 14; // STORE THE ACTUAL LENGTH OF MESSAGE INTO VARIABLE
+    qDebug() << "Length of message: ";
+    qDebug() << length_of_message;
+
+    quint8 frame_type = encoded_msg[3]; // 90 or 8B
+    qDebug() << "Frame type: ";
+    qDebug() << frame_type; // 90 -> 144, 8B -> 139 (to binary)
```

Comment:
Leaving qDebug() output function for testing output, will remove after testing decoding works